pennylane_lightning_kokkos/lightning_kokkos.py

```diff
@@ -245,12 +247,26 @@ def probability(self, wires=None, shot_range=None, bin_size=None):
     def expval(self, observable, shot_range=None, bin_size=None):
         if observable.name in [
             "Projector",
-            "Hamiltonian",
-            "SparseHamiltonian",
         ]:
             self.syncD2H()
             return super().expval(observable, shot_range=shot_range, bin_size=bin_size)
 
+        if observable.name in ["SparseHamiltonian"] and len(self.wires) >= 13:
```

Comment:
```suggestion
        if observable.name in ["SparseHamiltonian"]:
```

We shouldn't care too much about the wire size here. This should only be used for dense Hamiltonians. If we are expecting more than 13 wires we convert the dense H to a sparse H and use the above methods.